### PR TITLE
Support for basic if/elif directives

### LINKED ISSUE #15: Code in IfDirective can't currently be formatted
Given the following IfDirective, roslyn does not parse the contents because DEBUG is not defined
```c#
#if DEBUG
        Console.WriteLine("Rosyln does not parse this because DEBUG is not defined");
#endif
```
The contents are just a block of text.
Can I determine some way to parse those contents and format them? Currently they are just written out as is.

### PR DESCRIPTION
**Pull request by belav**
closes #15

This handles the basics of formatting code in if directives and supplies a configuration option for more complex cases.
#404 will improve upon this initial work

### CODE REVIEW

**Comment by belav** [comment]
I tested this across the whole of https://github.com/belav/csharpier-repos and came across

- An unrelated issue with switch/pattern matching that just happened to be in an `#if` #408
- An edge case that feels low priority #410
- Instances of files that won't compile because they have bad code inside of an `#if`

There may be some other edge cases similar to #410, I haven't reviewed the actual formatting changes this makes except on the aspnetcore repo. But I think this can be deemed good enough for the first pass.

File: Src/CSharpier.Tests/ConfigurationFileOptionsTests.cs [hunk 1]
```diff
@@ -36,22 +37,43 @@ public void Should_Return_Default_Options_With_No_File()
...
+                @"
+printWidth: 10
+preprocessorSymbolSets: 
+  - 
+    - 1
+    - 2
+  - 
+    - 3
+"
+            );
```
Comment (belav) [inline Src/CSharpier.Tests/ConfigurationFileOptionsTests.cs:70]: this feels ugly. Should it use comma delimited strings instead? 
```
preprocessorSymbolSets:
    - DEBUG,INCLUDE_X
    - INCLUDE_Y
```
vs
```
preprocessorSymbolSets:
    - 
        - DEBUG
        - INCLUDE_X
    - 
        - INCLUDE_Y
```
